src/core/smtp/MCSMTPSession.cpp

```diff
@@ -717,25 +745,15 @@ void SMTPSession::internalSendMessage(Address * from, Array * recipients, Data *
         mailsmtp_quit(mSmtp);
     }
     esmtp_address_list_free(address_list);
-
-    String * response;
-    int responseCode;
-
-    response = NULL;
-    if (mSmtp->response != NULL) {
-        response = String::stringWithUTF8Characters(mSmtp->response);
-        MC_SAFE_REPLACE_COPY(String, mLastSMTPResponse, response);
-    }
-    responseCode = mSmtp->response_code;
-    mLastSMTPResponseCode = responseCode;
+    saveLastResponse();
 
     if ((r == MAILSMTP_ERROR_STREAM) || (r == MAILSMTP_ERROR_CONNECTION_REFUSED)) {
         * pError = ErrorConnection;
         mShouldDisconnect = true;
         goto err;
     }
     else if (r == MAILSMTP_ERROR_EXCEED_STORAGE_ALLOCATION) {
-        if ((response != NULL) && (response->locationOfString(MCSTR("5.7.0")) != -1)) {
+        if ((mLastSMTPResponse != NULL) && (mLastSMTPResponse->locationOfString(MCSTR("5.7.0")) != -1)) {
```

Comment:
ha, my bad, it was still used here! Anyway, it's good either way. I'm merging the PR.